blazar/utils/trusts.py

```diff
@@ -58,7 +58,9 @@ def create_ctx_from_trust(trust_id):
         project_id=session.get_project_id(),
         service_catalog=(
             ctx.service_catalog or
-            session.auth.get_auth_ref().service_catalog),
+            session.auth.get_auth_ref(
+                session=session
+            ).service_catalog._catalog),
```

Comment:
service_catalog._catalog?

I think there's a property accessor you can use: https://github.com/openstack/keystoneauth/blob/112bcae1fbec355fcb58da07ae1d9f1adf8b77ba/keystoneauth1/access/service_catalog.py#L38-L46 -- actually this one looks like it's even better https://github.com/openstack/keystoneauth/blob/112bcae1fbec355fcb58da07ae1d9f1adf8b77ba/keystoneauth1/access/service_catalog.py#L101

So this doesn't want the `ServiceCatalog` type, it wants the dict structure?

Reply:
yes. it wants a dictionary.